dlib/overloaded.h

```diff
@@ -107,6 +113,7 @@ namespace dlib
 
         assert(type_ids == vector<int>({0,1,2}));
     !*/
+    noexcept(std::is_nothrow_constructible<detail::overloaded_helper<std::decay_t<T>...>, T&&...>::value)
```

Comment:
Nice, this is cool.

In the future though, if you put this into its own PR and send me that it will be easier.  Like you can stack branches and then send me this one change to overloaded and I would merge it more quickly.  Bigger PRs take more of my time to review so it ends up taking longer.  It's just easier for me to find time.  Like I'm an hour into reviewing this today and it's the middle of the night and I'm still looking it all over. So it won't get merged today.  But if it was in its own PR I would totally merge this bit right now (along with some of the other bits that could be separate).

Anyway, just saying :)  You would get faster turn around with more smaller PRs rather than fewer big ones.

Reply:
Really sorry. I'll endeavour to do this in the future.

Reply:
Na it's cool.  You do good work, don't sweat it :)

Just saying it will get your PRs merged faster.  No shame or anything like that :)

Reply:
I have to say, this kind of stuff illustrates what's wrong with C++. We have to manually tag stuff as `[[nodiscard]]`, `noexcept`, `constexpr`, etc when I'm willing to bet a modern compiler can determine all of this automatically without code having to be explicit about it. I don't think it's a matter of C++ having bad defaults, it's a matter of we shouldn't have to specify this in the first place. It should be automatically deduced. It just makes code unreadable and I doubt it's changing the generated assembly code at all. however, it's "best practice"...